### PR TITLE
Mini aodvalidation:  V0

### PR DESCRIPTION
**Pull request by nancymarinelli**
A new analyzer has been deployed in Validation/RecoEgamma and a new sequence to be included in Configuration/StandardSequences/python/Validation_cff.py (also committed). The machinery is meant only for MC miniAOD. It has all been tested with "central" running using runTheMatrix.py --what standard -i all -l 1332 --command=' -n 200'

### CODE REVIEW

**Comment by deguio** [comment]
please test

**Comment by nancymarinelli** [comment]
Hi Federico,
This one is for the 76X stream. I assume that I should make another PR for including the same stuff in the 74X stream ?

  Nancy
______________________________________

Nancy Marinelli

Research Associate Professor
University of Notre Dame, IN, US

CERN, Bdg 40/3-A01, 1211 Geneva,

SWITZERLAND

Phone +41-22-76-70809, fax +41-22-76-78940

**Comment by nancymarinelli** [comment]
Hello
The errors shown below have nothing to do with my additions/changes
I have hardly something to do with it. When I tested locally with runTheMatrix all was fine.

 Nancy
______________________________________

Nancy Marinelli

Research Associate Professor
University of Notre Dame, IN, US

CERN, Bdg 40/3-A01, 1211 Geneva,

SWITZERLAND

Phone +41-22-76-70809, fax +41-22-76-78940

**Comment by smuzaffar** [comment]
please test

File: Validation/RecoEgamma/plugins/PhotonValidatorMiniAOD.h [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  // Information from Particle Flow
+  // Isolation
+  MonitorElement* h_chHadIso_[3];
+  MonitorElement* h_nHadIso_[3];
+  MonitorElement* h_phoIso_[3];
+
+
+
+
+class  sortPhotons
```
Comment (deguio) [inline Validation/RecoEgamma/plugins/PhotonValidatorMiniAOD.h]: is this one needed?

File: Validation/RecoEgamma/plugins/PhotonValidatorMiniAOD.cc [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+  histname = "photonIso";
+  h_phoIso_[0]=  iBooker.book1D(histname+"All_miniAOD",   "PF photonIso:  All Ecal",etBin,etMin,20.);
+  h_phoIso_[1]=  iBooker.book1D(histname+"Barrel_miniAOD",   "PF photonIso:  Barrel",etBin,etMin,20.);
+  h_phoIso_[2]=  iBooker.book1D(histname+"Endcap_miniAOD",   "PF photonIso:  Endcap",etBin,etMin,20.);
+
+
+
+}
+
+void  PhotonValidatorMiniAOD::endRun (edm::Run const& r, edm::EventSetup const & theEventSetup) {}
```
Comment (deguio) [inline Validation/RecoEgamma/plugins/PhotonValidatorMiniAOD.cc]: please remove

File: Validation/RecoEgamma/plugins/PhotonValidatorMiniAOD.h [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+  //
+  explicit PhotonValidatorMiniAOD( const edm::ParameterSet& ) ;
+  virtual ~PhotonValidatorMiniAOD();
+
+
+  virtual void analyze( const edm::Event&, const edm::EventSetup& ) override;
+  //  virtual void beginJob();
+  virtual void dqmBeginRun( edm::Run const & r, edm::EventSetup const & theEventSetup) override;
+  virtual void endRun (edm::Run const& r, edm::EventSetup const & es) override;
```
Comment (deguio) [inline Validation/RecoEgamma/plugins/PhotonValidatorMiniAOD.h]: please remove

File: Validation/RecoEgamma/plugins/PhotonValidatorMiniAOD.cc (hunk 2 above)
Comment (nancymarinelli) [inline Validation/RecoEgamma/plugins/PhotonValidatorMiniAOD.cc]: I beg your pardon ??
I need these histograms. I am not sure what you're asking here.

 N.

______________________________________

Nancy Marinelli

Research Associate Professor
University of Notre Dame, IN, US

CERN, Bdg 40/3-A01, 1211 Geneva,

SWITZERLAND

Phone +41-22-76-70809, fax +41-22-76-78940

**Comment by deguio** [comment]
the comment was on line 183. please remove the endRun method and the sortPhotons class which is not used.
ciao and thanks,
F.

**Comment by nancymarinelli** [comment]
Ok. I will submit soonish a new PR with some addition to the analysis of the MiniAOD. I would wait for the new PR  for the removal of the redundant stuff, if you agree.

  Nancy
________________________________

**Comment by deguio** [comment]
you can update this one directly. it is probably easy.

**Comment by deguio** [comment]
-1
IIUC more development will come on top of this.
F.

**Comment by nancymarinelli** [comment]
Hi Federico
Sorry why –1 ??

It was tsetd and the development will come later. This needs to go in production the sooner
Thanks,
 Nancy
______________________________________

Nancy Marinelli

Research Associate Professor
University of Notre Dame, IN, US

CERN, Bdg 40/3-A01, 1211 Geneva,

SWITZERLAND

Phone +41-22-76-70809, fax +41-22-76-78940

**Comment by deguio** [comment]
+1
ok, we discussed. new dev will come with a different PR.

**Comment by deguio** [comment]
removed my approva. keeping on hold since it seems it causes problem in 80. let's solve there first.
F.

**Comment by davidlange6** [comment]
I am cleaning up the 76x queue aside from things for analysis workflows. I'm closing this pull request, please make sure the PR is in 80x. Thanks!

**Pull request closed without merging**